types/geojson/index.d.ts

```diff
@@ -1,123 +1,157 @@
+     * https://tools.ietf.org/html/rfc7946#section-5
+     */
+    bbox?: BoundingBox;
 }
+
 /**
- * GeometryObject supports geometry collection as well
+ * A geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3
  */
-export type GeometryObject = DirectGeometryObject | GeometryCollection;
+export interface GeometryObject extends GeoJsonObject {
+    type: GeoJsonGeometryTypeNames;
+}
 
-/***
- * http://geojson.org/geojson-spec.html#point
+/**
+ * Point geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.2
  */
-export interface Point extends DirectGeometryObject {
+export interface Point extends GeometryObject {
     type: "Point";
     coordinates: Position;
 }
 
-/***
- * http://geojson.org/geojson-spec.html#multipoint
+/**
+ * MultiPoint geometry object.
+ *  https://tools.ietf.org/html/rfc7946#section-3.1.3
  */
-export interface MultiPoint extends DirectGeometryObject {
+export interface MultiPoint extends GeometryObject {
     type: "MultiPoint";
     coordinates: Position[];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#linestring
+/**
+ * LineString geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.4
  */
-export interface LineString extends DirectGeometryObject {
+export interface LineString extends GeometryObject {
     type: "LineString";
     coordinates: Position[];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#multilinestring
+/**
+ * MultiLineString geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.5
  */
-export interface MultiLineString extends DirectGeometryObject {
+export interface MultiLineString extends GeometryObject {
     type: "MultiLineString";
     coordinates: Position[][];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#polygon
+/**
+ * Polygon geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.6
  */
-export interface Polygon extends DirectGeometryObject {
+export interface Polygon extends GeometryObject {
     type: "Polygon";
     coordinates: Position[][];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#multipolygon
+/**
+ * MultiPolygon geometry object.
+ * https://tools.ietf.org/html/rfc7946#section-3.1.7
  */
-export interface MultiPolygon extends DirectGeometryObject {
+export interface MultiPolygon extends GeometryObject {
     type: "MultiPolygon";
     coordinates: Position[][][];
 }
 
-/***
- * http://geojson.org/geojson-spec.html#geometry-collection
+/**
+ * Geometry Collection
+ * https://tools.ietf.org/html/rfc7946#section-3.1.8
  */
-export interface GeometryCollection extends GeoJsonObject {
+export interface GeometryCollection extends GeometryObject {
     type: "GeometryCollection";
     geometries: GeometryObject[];
 }
 
-/***
+/**
+ * A feature object which contains a geometry and associated properties.
  * https://tools.ietf.org/html/rfc7946#section-3.2
  */
-export interface Feature<G extends GeometryObject, P = any> extends GeoJsonObject {
```

Comment:
👎 👎 Disapprove this change, how would one define a `Feature<LineString>`?? This makes the Type more generalized and also it would break EVERY GeoJSON type definition.

Reply:
Seconded from the perspective of the **d3-geo** definitions. The generics should not be dropped. 👎

Reply:
I would also not drop the the generics, but I think it would be OK to make something like this:

```typescript
type GeoJSONProperties = { [name: string]: any; } | null;
export interface Feature<G extends GeometryObject, P = GeoJSONProperties> extends GeoJsonObject {
    // ...
}
```

In other words: Keep both generics, but make the definition of the `properties` more strict...

Reply:
👍 @atd-schubert's `GeoJSONProperties` type